### PR TITLE
Move FIFOCompactionPicker to a separate file

### PR DESCRIPTION
**Pull request by sagar0**
**Summary:**
Simplified the code layout by moving FIFOCompactionPicker to a separate file.
**Why?:**
While trying to add ttl functionality to universal compaction, I found that `FIFOCompactionPicker` class and its impl methods to be interspersed between `LevelCompactionPicker` methods which kind-of made the code a little hard to traverse. So I moved `FIFOCompactionPicker` to a separate compaction_picker_fifo.h/cc file, similar to `UniversalCompactionPicker`. 

Test Plan:
`make check`

### CODE REVIEW

File: tools/db_bench_tool.cc [hunk 1]
```diff
@@ -4580,7 +4580,8 @@ void VerifyDBFromDB(std::string& truth_db_name) {
       if (FLAGS_max_scan_distance != 0) {
         if (FLAGS_reverse_iterator) {
           GenerateKeyFromInt(
-              (uint64_t)std::max((int64_t)0, seek_pos - FLAGS_max_scan_distance),
+              (uint64_t)std::max((int64_t)0,
+                                 seek_pos - FLAGS_max_scan_distance),
```
Comment (sagar0) [inline tools/db_bench_tool.cc]: This change is unrelated to this PR and got added by `make format`. But I think it might be ok to keep it in this PR as it is just a minor formatting fix.

File: db/compaction_picker.cc [hunk 2]
```diff
@@ -142,6 +110,39 @@ void CompactionPicker::ReleaseCompactionFiles(Compaction* c, Status status) {
   }
 }
 
+bool CompactionPicker::FindIntraL0Compaction(
```
Comment (riversand963) [inline db/compaction_picker.cc]: Is `FindIntraL0Compaction` the same as before except that it becomes a member function? Does it need access to any of the members of `CompactionPicker`? If not, maybe it does not have to be a member function. Let me know if you have other thoughts.

Comment (sagar0) [reply db/compaction_picker.cc]: Thanks for reviewing, @riversand963. 

`FindIntraL0Compaction` is the same as before, and does not need access to any fields of `CompactionPicker`. I guess you have a good point that I need not make it a member function. I only made it a member of CompactionPicker class as `FIFOCompactionPicker` also makes use of it, in addition to Level. 
Let me try moving it back out of `CompactionPicker`.

**Comment by sagar0** [comment]
@riversand963 can you take another look? I updated based on your suggestion.
